cpp/src/arrow/datum.h

```diff
@@ -283,6 +283,8 @@ struct ARROW_EXPORT Datum {
   std::string ToString() const;
 
   ARROW_EXPORT friend void PrintTo(const Datum&, std::ostream*);
+
+  ARROW_EXPORT std::string ToString(Datum::Kind kind);
```

Comment:
Why is this a method of Datum and not a free function?

Reply:
We can do that.